src/lib.rs

```diff
@@ -1,576 +1,333 @@
-                .expect("invalid delegation");
-            assert_eq!(decoded, rcan_2);
-        }
-    }
-
-    #[test]
-    fn test_simple_caps() {
-        let mut rng = ChaCha8Rng::seed_from_u64(0);
-
-        let issuer = SigningKey::generate(&mut rng);
-        let receiver = SigningKey::generate(&mut rng);
-
-        let mut caps = Caps::<MyCap>::default();
-
-        // The issuer delegates these caps
-        let write_app_read_api_cap = MyCapBuilder::new()
-            .with_api(CapState::Read)
-            .with_app(CapState::Write)
-            .build();
-        caps.insert(issuer.verifying_key(), write_app_read_api_cap);
-
-        let rcan = Rcan::new(&mut rng, &issuer, receiver.verifying_key(), caps);
-
-        let encoded = rcan.encode_to_bytes();
-        let decoded = Rcan::decode(&encoded).expect("failed to verify");
-
-        assert_eq!(rcan, decoded);
-
-        assert!(MyCapBuilder::new()
-            .with_api(CapState::Read)
-            .build()
-            .can_delegate_to_us(&MyCapBuilder::new().with_api(CapState::Read).build()));
-
-        assert!(MyCapBuilder::new()
-            .with_api(CapState::Read)
-            .build()
-            .can_delegate_to_us(&MyCapBuilder::new().with_api(CapState::Write).build()));
+    fn test_simple_attenuations() {
+        assert!(Rpc::Read.allowed_by_attenuation(&Rpc::Read));
+        assert!(!Rpc::ReadWrite.allowed_by_attenuation(&Rpc::Read),);
+        assert!(Rpc::ReadWrite.allowed_by_attenuation(&Rpc::ReadWrite),);
     }
 
     #[test]
-    fn test_expired() {
-        let mut rng = ChaCha8Rng::seed_from_u64(0);
-
-        let issuer = SigningKey::generate(&mut rng);
-        let receiver = SigningKey::generate(&mut rng);
-
-        let caps = Caps::<()>::default();
-        let now = now();
-
-        let rcan =
-            Rcan::new_with_expiry(&mut rng, &issuer, receiver.verifying_key(), caps, now + 1);
-
-        let encoded = rcan.encode_to_bytes();
-
-        // valid
-        let _ = Rcan::<()>::decode_with_time(&encoded, now).expect("should verify");
-
-        // expired
-        let err = Rcan::<()>::decode_with_time(&encoded, now + 3).unwrap_err();
-        assert!(err.to_string().contains("expired"), "{}", err);
+    fn test_rcan_encoding() -> TestResult {
+        let issuer = SigningKey::from_bytes(&[0u8; 32]);
+        let audience = SigningKey::from_bytes(&[1u8; 32]);
+        let rcan = Rcan::builder(&issuer, audience.verifying_key())
+            .issuing(Rpc::ReadWrite)
+            .sign(Expires::Never);
+
+        println!("{}", hex::encode(rcan.encode()));
+        assert_eq!(hex::encode(rcan.encode()), "01203b6a27bcceb6a42d62a3a8d02a6f0d73653215771de243a63ac048a18b59da29208a88e3dd7409f195fd52db2d3cba5d72ca6709bf1d94121bf3748801b40f6f5c01203b6a27bcceb6a42d62a3a8d02a6f0d73653215771de243a63ac048a18b59da290100cede73ce610c4196671a60335beca23464f06cbc9402bf1c9b7377e36453c43d2652fd2c857fab4300fa92da6ea357435c341123ef89d189884e56f58ce88206");
```

Comment:
I wanted to use the hexdump utility here at some point to make sure the encoding format doesn't encode unnecessary gack.
I want to prevent e.g. classic cases where a `Vec<u8>` is serialized as a list of individual numbers, stuff like that.